### PR TITLE
circleCI: build docker image with latest gometalinter release

### PR DESCRIPTION
**Pull request by nhoriguchi**
We can reduce hard-coding a version in Dockerfile, which will make code maintenance a bit easy.

Based on discussion in #52, and original issue is #51.

### CODE REVIEW

File: .circleci/images/Dockerfile [hunk 1]
```diff
@@ -18,7 +18,8 @@ RUN SGX_SDK_URL="https://download.01.org/intel-sgx/linux-2.3.1/ubuntu18.04/sgx_l
   && rm -f /tmp/sgx_linux_x64_sdk.bin \
   && sudo ln -s /opt/intel/sgxsdk/environment /etc/profile.d/intel-sgxsdk.sh
 
-RUN GOMETALINTER_URL="https://github.com/alecthomas/gometalinter/releases/download/v2.0.11/gometalinter-2.0.11-linux-amd64.tar.gz" \
-  && sudo mkdir /opt/gometalinter \
-  && curl -L $GOMETALINTER_URL | sudo tar --strip-components=1 -C /opt/gometalinter -xzf -
+COPY install_latest_gometalinter.sh /tmp/install_latest_gometalinter.sh
+RUN sudo apt-get install -y jq \
+  && bash /tmp/install_latest_gometalinter.sh
```
Comment (ynamiki) [inline .circleci/images/Dockerfile:23]: Is it possible to write contents of the shell script here directly? I am wondering why there is a separate shell script.

File: .circleci/images/Dockerfile [hunk 2]
```diff
@@ -18,7 +18,8 @@ RUN SGX_SDK_URL="https://download.01.org/intel-sgx/linux-2.3.1/ubuntu18.04/sgx_l
   && rm -f /tmp/sgx_linux_x64_sdk.bin \
   && sudo ln -s /opt/intel/sgxsdk/environment /etc/profile.d/intel-sgxsdk.sh
 
-RUN GOMETALINTER_URL="https://github.com/alecthomas/gometalinter/releases/download/v2.0.11/gometalinter-2.0.11-linux-amd64.tar.gz" \
-  && sudo mkdir /opt/gometalinter \
-  && curl -L $GOMETALINTER_URL | sudo tar --strip-components=1 -C /opt/gometalinter -xzf -
+COPY install_latest_gometalinter.sh /tmp/install_latest_gometalinter.sh
+RUN sudo apt-get install -y jq \
```
Comment (ynamiki) [inline .circleci/images/Dockerfile:22]:
```suggestion
RUN sudo apt-get install -y \
    jq \
```

File: .circleci/images/Dockerfile (hunk 1 above)
Comment (nhoriguchi) [reply .circleci/images/Dockerfile:23]: Yes, firstly I considered that approach, but I felt it's hard to keep code readable when there's a line assigning variable from the output of other command. But anyway, I'll try to open it.

File: .circleci/images/Dockerfile (hunk 2 above)
Comment (nhoriguchi) [reply .circleci/images/Dockerfile:22]: You're right, this is stated as [one of best practices](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/).

**Comment by nhoriguchi** [comment]
Updated version is uploaded, so close this issue.

**Pull request closed without merging**